### ISSUE TITLE
MSAL lockfile

### ISSUE THREAD

**Issue by isaiahwilliams** [microsoft/secmgmt-open-powershell#5]
# Steps to reproduce

Invoke the following 

```powershell
Connect-SecMgmtAccount
```

## Expected behavior

Expecting successful authentication

## Actual behavior

```output
Connect-SecMgmtAccount : Could not get access to the shared lock file.
At line:1 char:1
+ Connect-SecMgmtAccount -Debug
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Connect-SecMgmtAccount], InvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Online.SecMgmt.PowerShell.Commands.ConnectSecMgmtAccount
```

## Diagnostic logs

The `msal.cache.lockfile` is not being released which is blocking all requests for access tokens

```output
Exception type: System.IO.IOException

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights,
FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean
bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize,
FileOptions options)
   at Microsoft.Identity.Client.Extensions.Msal.CrossPlatLock..ctor(String lockfilePath, Int32 lockFileRetryDelay,
Int32 lockFileRetryCount)
=== End of inner exception stack trace ===

   at Microsoft.Identity.Client.Extensions.Msal.CrossPlatLock..ctor(String lockfilePath, Int32 lockFileRetryDelay,
Int32 lockFileRetryCount)
   at Microsoft.Identity.Client.Extensions.Msal.MsalCacheHelper.CreateCrossPlatLock(StorageCreationProperties
storageCreationProperties)
   at Microsoft.Identity.Client.Extensions.Msal.MsalCacheHelper.<CreateAsync>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Online.SecMgmt.PowerShell.Utilities.PersistentTokenCache.GetMsalCacheStorage()
```

## Environment

- PowerShell 5.1
- Windows 10 2004

**Comment by isaiahwilliams** [comment]
Root cause for this issue is a change in the newer versions of `Microsoft.Identity.Client.Extensions.Msal`. When creating a `MsalCacheHelper` object it attempts to create a lock file, and this attempt will fail because one already exists due to the following code 

```csharp
        public override void BeforeAccessNotification(TokenCacheNotificationArgs args)
        {
            MsalCacheHelper cacheStorage = GetMsalCacheStorage();

            args.AssertNotNull(nameof(args));

            try
            {
                cacheLock = new CrossPlatformLock($"{CacheFilePath}.lockfile");

                cacheLock.CreateLockAsync().ConfigureAwait(false);
                args.TokenCache.DeserializeMsalV3(cacheStorage.LoadUnencryptedTokenCache());
            }
            catch (Exception)
            {
                cacheStorage.Clear();
                throw;
            }
        }
```

**Comment by isaiahwilliams** [comment]
This issue has been resolved by PR #6
